### ISSUE TITLE
Authentication error

### ISSUE THREAD

**Issue by Alexander-Hjelm** [solidify/jira-azuredevops-migrator#584]
### Discussed in https://github.com/solidify/jira-azuredevops-migrator/discussions/583

<div type='discussions-op-text'>

<sup>Originally posted by **royc75** October 24, 2022</sup>
Hello

I'm trying to use the tool on Jira server v8.20.12 which is using Microsoft MFA.

I'm getting the following 403 upon authentication:

```
  <div class="aui-message aui-message-warning warning">
                            <p>Encountered a <code>&quot;403 - Forbidden&quot;</code> error while loading this page.</p>
                            <p>Basic Authentication Failure - Reason : AUTHENTICATION_DENIED</p>
                            <p><a href="/secure/MyJiraHome.jspa">Go to Jira home</a></p>
                        </div>
```

I'm sure my credentials are correct.

I tried every combination I could think about - My user name with and without the domain, my actual password as well as access token. Same access token works for example for Visual Studio code Jira add on.

Any ideas are most welcome.

Thanks
Roy
</div>

**Comment by Alexander-Hjelm** [comment]
@[royc75, We don't support instances where you are running MFA on your Jira Server, AFAIK. The password should be your JIRA Token, and you need to have the JIRA Server open to REST API calls, otherwise it will not work :)

**Comment by royc75** [comment]
@Alexander-Hjelm REST is open.

VS Code Atlassian extension uses that as well and it works for me there with the same details.

**Comment by royc75** [comment]
@Alexander-Hjelm I have also tried with Jira admin user which has got access to everything, same error still. If that solution doesn't support MFA fair enough but most organisations use it these days.

**Comment by Alexander-Hjelm** [comment]
Shouldn't matter. You can make REST API calls using username and the API token generated by the app. That will bypass the MFA.

-u is your full email and -p is the Jira access token you generated?

Check: https://github.com/solidify/jira-azuredevops-migrator/issues/535

**Comment by royc75** [comment]
Yes it is and as I said the exact same username and token works for other integrations but not for this one.

**Comment by Alexander-Hjelm** [comment]
Could be something funky with CAPTCHA on the JIRA server side.

I know that you can set an optional SecurityProtocol here: https://github.com/solidify/jira-azuredevops-migrator/blob/master/src/WorkItemMigrator/JiraExport/JiraServiceWrapper.cs#L28. But we have never had the need to. MFA usually isn't a problem. You could maybe play around with that and see if you can get it working.

**Comment by royc75** [comment]
Tried, no luck but thanks....

![image](https://user-images.githubusercontent.com/40601216/198535230-1cebf902-0d91-43e2-9065-9d35f2f49fc6.png)

**Comment by Alexander-Hjelm** [comment]
No worries: We'll track this as a bug in case something fishy is going on, but can't say when we'll get back to it :)

In the meantime, a bypass solution could be to move a snapshot of your Jira Project to cloud with the Cloud Migration Assistant (https://confluence.atlassian.com/jirakb/how-to-migrate-a-project-from-jira-server-to-jira-cloud-779160760.html), and then use our tool to set up a migration job from Jira Cloud to ADO.

**Comment by royc75** [comment]
Thanks ! But we don't have Jira Cloud subscription :)

**Comment by Alexander-Hjelm** [comment]
They have a free jira cloud plan now, I think. Don't know if it's relevant for you, but still an option :) https://support.atlassian.com/jira-cloud-administration/docs/what-is-the-free-jira-cloud-plan/

**Comment by royc75** [comment]
Works with Jira cloud so must be our setup :-/

**Comment by Alexander-Hjelm** [comment]
@royc75, we have the option of authenticating via Oauth instead. Would such a solution work for you? and would it be possible for you to provide all the secrets and tokens as shown below:

![image](https://user-images.githubusercontent.com/10683896/199470913-9e06ec98-0697-4207-a48d-b41bb6fb35a8.png)

**Comment by roycohen1** [comment]
@Alexander-Hjelm Afraid not, I'm not the master of my own domain in this organisation but thanks for the suggestion !

I'm still chasing the admins to advise as I believe it is due to policy / permissions in our Jira

**Comment by Alexander-Hjelm** [comment]
Ok! Let me know if you figure something out! I'll try to look into the issue as best I can in the meantime.

**Comment by royc75** [comment]
@Alexander-Hjelm The Jira team has created a service account for me which bypass MFA for SSO and it now works ! Unrelated question and apologise for hijacking this discussion for it: Is there a way to migrate work items such that Jira sprints mirror ADO iterations ?

**Comment by Alexander-Hjelm** [comment]
@royc75 Ok, good to know! I'll recommend this as the first solution to try if anybody experiences the same problem again.

For the sprints, you will have to set the `sprint-field` property in your config, and add a `SprintMapper`. See this config for an example: https://github.com/solidify/jira-azuredevops-migrator/blob/master/docs/config.md

**Comment by royc75** [comment]
Thanks @Alexander-Hjelm I've got few other questions. Would it be possible to connect outside github please ?
